File: app_manager/appmanager/src/main/java/app/editors/manager/ui/activities/main/ProfileActivity.java
package app.editors.manager.ui.activities.main;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

import app.editors.manager.R;
import app.editors.manager.mvp.models.account.AccountsSqlData;
import app.editors.manager.ui.activities.base.BaseAppActivity;
import app.editors.manager.ui.fragments.main.ProfileFragment;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import lib.toolkit.base.managers.utils.FragmentUtils;

public class ProfileActivity extends BaseAppActivity {

    public static final String TAG = ProfileActivity.class.getSimpleName();
    public static final int REQUEST_PROFILE = 1001;

    @BindView(R.id.appToolbar)
    Toolbar mToolbar;

    private Unbinder mUnbinder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_profile);
        mUnbinder = ButterKnife.bind(this);
        init();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mUnbinder != null) {
            mUnbinder.unbind();
        }
    }

    private void init() {
        initToolbar();
        showFragment();
    }

    private void initToolbar() {
        setSupportActionBar(mToolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setTitle(R.string.fragment_profile_title);
            mToolbar.setNavigationOnClickListener(v -> onBackPressed());
        }
    }

    private void showFragment() {
        Fragment fragment = getSupportFragmentManager().findFragmentByTag(ProfileFragment.TAG);
        if (fragment == null) {
            fragment = ProfileFragment.newInstance(getIntent().getParcelableExtra(ProfileFragment.KEY_ACCOUNT));
        }
        FragmentUtils.showFragment(getSupportFragmentManager(), fragment, R.id.frame_container);
    }


    public static void show(Activity activity, AccountsSqlData account) {
        final Intent intent = new Intent(activity, ProfileActivity.class);
        intent.putExtra(ProfileFragment.KEY_ACCOUNT, account);
        activity.startActivityForResult(intent, REQUEST_PROFILE);
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/activities/main/ProfileActivity.kt
package app.editors.manager.ui.activities.main

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import app.editors.manager.R
import app.editors.manager.ui.activities.base.BaseAppActivity
import app.editors.manager.ui.fragments.main.ProfileFragment
import lib.toolkit.base.managers.utils.FragmentUtils

class ProfileActivity : BaseAppActivity() {

    companion object {
        val TAG: String = ProfileActivity::class.java.simpleName

        const val REQUEST_PROFILE = 1001
        const val RESULT_LOGIN = 1005

        fun show(activity: Activity, account: String) {
            activity.startActivityForResult(Intent(activity, ProfileActivity::class.java).apply {
                putExtra(ProfileFragment.KEY_ACCOUNT, account)
            }, REQUEST_PROFILE)
        }
    }


    private lateinit var toolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_profile)
        init(savedInstanceState)
    }

    private fun init(savedInstanceState: Bundle?) {
        initToolbar()
        if (savedInstanceState == null) {
            showFragment()
        }
    }

    private fun initToolbar() {
        toolbar = findViewById(R.id.appToolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.let {
            it.setDisplayHomeAsUpEnabled(true)
            it.setTitle(R.string.fragment_profile_title)
            toolbar.setNavigationOnClickListener { onBackPressed() }
        }
    }

    private fun showFragment() {
        val fragment = ProfileFragment.newInstance(intent.getStringExtra(ProfileFragment.KEY_ACCOUNT))
        FragmentUtils.showFragment(supportFragmentManager, fragment, R.id.frame_container)
    }

}